example/synchronous_resnet/resnet_worker.py

```diff
@@ -0,0 +1,566 @@
+        else:
+            sub_net, parent_layer_name = build_residual_block(net[parent_layer_name],
+                                                              1.0/4, 1, False, 4, ix='3%s' % c)
+        net.update(sub_net)
+
+    block_size = list('abcdef')
+    for c in block_size:
+        if c == 'a':
+            sub_net, parent_layer_name = build_residual_block(net[parent_layer_name],
+                                                              1.0/2, 1.0/2, True, 4, ix='4%s' % c)
+        else:
+            sub_net, parent_layer_name = build_residual_block(net[parent_layer_name],
+                                                              1.0/4, 1, False, 4, ix='4%s' % c)
+        net.update(sub_net)
+
+    block_size = list('abc')
+    for c in block_size:
+        if c == 'a':
+            sub_net, parent_layer_name = build_residual_block(net[parent_layer_name],
+                                                              1.0/2, 1.0/2, True, 4, ix='5%s' % c)
+        else:
+            sub_net, parent_layer_name = build_residual_block(net[parent_layer_name],
+                                                              1.0/4, 1, False, 4, ix='5%s' % c)
+        net.update(sub_net)
+
+    net['pool5'] = PoolLayer(net[parent_layer_name], pool_size=7, stride=1, pad=0,
+                             mode='average_exc_pad', ignore_border=False)
+    net['fc1000'] = DenseLayer(net['pool5'], num_units=1000, nonlinearity=None)
+    net['prob'] = NonlinearityLayer(net['fc1000'], nonlinearity=softmax)
+
+    return net
+
+
+def split_data_for_worker(dataset, splits, name):
+    data = dataset[0][splits[name][0]:splits[name][1]]
+    targets = dataset[1][splits[name][0]:splits[name][1]]
+    return [data, targets]
+
+
+def pred_error(f_pred, data, iterator):
+    """
+    Just compute the error
+    f_pred: Theano fct computing the prediction
+    """
+    valid_err = 0.
+    i = 0
+    for _, valid_index in iterator:
+        x = [data[0][t] for t in valid_index]
+        y = [data[1][t] for t in valid_index]
+        valid_err += f_pred(x, y)
+        i += 1
+
+    return valid_err / i
+
+
+def train_resnet(
+    batch_size=12,  # The batch size during training.
+    valid_batch_size=12,  # The batch size used for validation/test set.
+    validFreq=5,
+    lrate=1e-4,
+    optimizer=sgd,
+):
+
+    # Each worker needs the same seed in order to draw the same parameters.
+    # This will also make them shuffle the batches the same way, but splits are
+    # different so doesnt matter
+    seed = worker.send_req('seed')
+    numpy.random.seed(seed)
+
+    print('Loading data')
+    train, valid, test = load_data()
+
+    print('Building model')
+    resnet = build_resnet()
+    params = lasagne.layers.get_all_params(resnet.values(), trainable=True)
+
+    print("Using all_reduce worker's interface!")
+    asgd = AverageSGD(worker)
+    asgd.make_rule(params)
```

Comment:
Can you please add other update algorithm, like `EASGD`, `Downpour` around here, so we can see the differences please?